pkg/controllers/cspi-controller/handler.go

```diff
@@ -195,40 +195,26 @@ func (c *CStorPoolInstanceController) updateStatus(cspi *cstor.CStorPoolInstance
 	var status cstor.CStorPoolInstanceStatus
 	var err error
 	pool := zpool.PoolName()
+	propertyList := []string{"health", "io.openebs:readonly"}
 
-	state, er := zpool.GetPropertyValue(pool, "health")
-	if er != nil {
-		err = zpool.ErrorWrapf(err, "Failed to fetch health")
+	valueList, er := zpool.GetListOfPropertyValues(pool, propertyList)
+	if er != nil || len(valueList) != len(propertyList) {
+		return errors.Wrapf(err, "Failed to fetch %v output: %v", propertyList, valueList)
```

Comment:
Is the `valueList` and  `propertyList` in order ? 
I mean if let say , propertyList is -- ["Prop1", "Prop2"]
Will the valueList be in respective order -- ["valueForProp1", "valueForProp2"] ?

Reply:
Yes!! It will be queried in the same order

Reply:
and output also will be in the same order